util/src/main/scala/org/ensime/vfs/EnsimeVFS.scala

```diff
@@ -46,6 +46,22 @@ object `package` {
   private val zipFileField = classOf[ZipFileSystem].getDeclaredField("zipFile")
   zipFileField.setAccessible(true)
 
+  object URLParamEncoder {
+    def encode(input: String): String = {
```

Comment:
have you done an object allocation profile of that? I bet it's not pretty! :laughing: